third_party/terraform/resources/resource_logging_sink.go

```diff
@@ -56,21 +73,39 @@ func flattenResourceLoggingSink(d *schema.ResourceData, sink *logging.LogSink) {
 	d.Set("writer_identity", sink.WriterIdentity)
 }
 
-func expandResourceLoggingSinkForUpdate(d *schema.ResourceData) *logging.LogSink {
+func expandResourceLoggingSinkForUpdate(d *schema.ResourceData) (*logging.LogSink, string) {
 	// Can only update destination/filter right now. Despite the method below using 'Patch', the API requires both
 	// destination and filter (even if unchanged).
 	sink := logging.LogSink{
-		Destination: d.Get("destination").(string),
-		Filter:      d.Get("filter").(string),
+		Destination:     d.Get("destination").(string),
+		Filter:          d.Get("filter").(string),
+		ForceSendFields: []string{"Destination", "Filter"},
```

Comment:
@chrisst 
If I understand and remember correctly (which _might not_ be the case), those two fields are required to be included in the request body for update even when they aren't updated. That's why I put them in `ForceSendField` statically.
On the other hand, `updateMask` is used to express which fields should be updated. This works even if specified fields are not included in the API request; e.g.: if you call update API with request body missing `bigqueryOptions` field but the field name `bigqueryOtions` appears in `updateMask`, the resource will be updated to drop BigQuery options.

Anyway, the above info might be wrong, so I'll write some test to examine them. Let me take my time (maybe updated within this week).

Reply:
By not specifying `usePartitionedTables` as a force send field Terraform will send ` "bigqueryOptions": {}` to the API when this field is marked as `false` as Terraform and the Go client don't have a good awareness of unset/false vs set/false booleans. However in this case it appears that the upstream API has the same problem and when `usePartitionedTables` is explicitly sent as false in the API it returns ` "bigqueryOptions": {}` in the API response. 

So while many API's would have required this field to be forced apparently this API works better without that. So you are correct that updateMask in this circumstance is adequate.

Reply:
Good explanation, so it's adequate just because it fits the behavior of the API. I'll keep the point in my mind and try to remember when I'll fix another resource in the future...